atc/atccmd/command.go

```diff
@@ -96,387 +99,252 @@ const algorithmLimitRows = 100
-	var newKey *encryption.Key
-	var oldKey *encryption.Key
-
-	if cmd.EncryptionKey.AEAD != nil {
-		newKey = encryption.NewKey(cmd.EncryptionKey.AEAD)
-	}
-	if cmd.OldEncryptionKey.AEAD != nil {
-		oldKey = encryption.NewKey(cmd.OldEncryptionKey.AEAD)
-	}
-
-	helper := migration.NewOpenHelper(
-		defaultDriverName,
-		cmd.Postgres.ConnectionString(),
-		cmd.lockFactory,
-		newKey,
-		oldKey,
-	)
-
-	version, err := helper.CurrentVersion()
-	if err != nil {
-		return err
-	}
-
-	return helper.MigrateToVersion(version)
-}
-
-func (cmd *Migration) migrateToLatestVersion() error {
-	helper := migration.NewOpenHelper(
-		defaultDriverName,
-		cmd.Postgres.ConnectionString(),
-		cmd.lockFactory,
-		nil,
-		nil,
-	)
-
-	version, err := helper.SupportedVersion()
-	if err != nil {
-		return err
-	}
-
-	return helper.MigrateToVersion(version)
-}
-
-func (cmd *ATCCommand) WireDynamicFlags(commandFlags *flags.Command) {
-	cmd.RunCommand.WireDynamicFlags(commandFlags)
-}
-
-func (cmd *RunCommand) WireDynamicFlags(commandFlags *flags.Command) {
-	var (
-		metricsGroup      *flags.Group
-		policyChecksGroup *flags.Group
-		credsGroup        *flags.Group
-		authGroup         *flags.Group
-	)
-
-	groups := commandFlags.Groups()
-	for i := 0; i < len(groups); i++ {
-		group := groups[i]
-
-		if credsGroup == nil && group.ShortDescription == "Credential Management" {
-			credsGroup = group
-		}
-
-		if metricsGroup == nil && group.ShortDescription == "Metrics & Diagnostics" {
-			metricsGroup = group
-		}
-
-		if policyChecksGroup == nil && group.ShortDescription == "Policy Checking" {
-			policyChecksGroup = group
-		}
-
-		if authGroup == nil && group.ShortDescription == "Authentication" {
-			authGroup = group
-		}
-
-		if metricsGroup != nil && credsGroup != nil && authGroup != nil && policyChecksGroup != nil {
-			break
-		}
+var CmdDefaults RunConfig = RunConfig{
+	Logger: flag.Lager{
+		LogLevel: "info",
+	},
+
+	BindIP:   net.IPv4(0, 0, 0, 0),
+	BindPort: 8080,
+
+	TLS: TLSConfig{},
+	Auth: AuthConfig{
+		AuthFlags: skycmd.AuthFlags{
+			Expiration: 24 * time.Hour,
+			Connectors: skycmd.ConnectorsConfig{
+				OAuth: skycmd.OAuthFlags{
+					GroupsKey:   "groups",
+					UserIDKey:   "user_id",
+					UserNameKey: "user_name",
+				},
 
-		groups = append(groups, group.Groups()...)
-	}
+				OIDC: skycmd.OIDCFlags{
+					GroupsKey:   "groups",
+					UserNameKey: "username",
+				},
 
-	if metricsGroup == nil {
-		panic("could not find Metrics & Diagnostics group for registering emitters")
-	}
+				SAML: skycmd.SAMLFlags{
+					UsernameAttr: "name",
+					EmailAttr:    "email",
+					GroupsAttr:   "groups",
+				},
+			},
+		},
+	},
+
+	Server: ServerConfig{
+		XFrameOptions: "deny",
+		ClientID:      "concourse-web",
+	},
+
+	SystemClaim: SystemClaimConfig{
+		Key:    "aud",
+		Values: []string{"concourse-worker"},
+	},
+
+	LetsEncrypt: LetsEncryptConfig{
+		ACMEURL: ignoreErrParseURL("https://acme-v02.api.letsencrypt.org/directory"),
```

Comment:
tiny suggestion: a convention I've seen for this is to call the function `mustParseURL` and rather than ignoring it'll panic on any err. this convention comes from go stdlib: https://golang.org/pkg/regexp/#MustCompile